cli/lightning-cli.c

```diff
@@ -265,11 +265,42 @@ static void print_json(const char *str, const jsmntok_t *tok, const char *indent
 	abort();
 }
 
+/* Always returns a positive number < len.  len must be > 0! */
+static size_t read_nofail(int fd, void *buf, size_t len)
+{
+	ssize_t i = read(fd, buf, len);
```

Comment:
Given the comment "len must be > 0" maybe add `assert(len > 0);`?

Reply:
Agreed, done.